### PR TITLE
add rust-analyzer support

### PR DESCRIPTION
**Pull request by jiacai2050**

### CODE REVIEW

File: core/lspserver.py [hunk 1]
```diff
@@ -347,7 +347,7 @@ def send_completion_request(self, request_id, filepath, type, row, column, char)
     def send_find_define_request(self, request_id, filepath, type, row, column):
         self.record_request_id(request_id, filepath, type)
 
-        self.send_to_request("textDocument/declaration",
+        self.send_to_request("textDocument/definition",
```
Comment (jiacai2050) [inline core/lspserver.py:350]: Since this method is `send_find_define_request`, it make more sense to send `definition` message.

**Comment by jiacai2050** [comment]
@manateelazycat I test with [this file](https://github.com/1history/1History/blob/74b5a1667c4f0e452beace35f8ce03fd6c64e5e8/src/database.rs#L26)

`lsp-bridge-find-define` works as expected, but not  for completion.
After typing `conn.`,  `company-diag` return 
```
Emacs 28.1 (aarch64-apple-darwin21.1.0) of 2022-04-05 on armbob.lan
Company 0.9.13

company-backends: ((company-lsp-bridge))

Used backend: (company-lsp-bridge)

Major mode: rust-mode
Prefix: ""
Completions: none
```

In `*lsp-bridge*` buffer, rust-analyzer return 174 completion items, I have no idea why company got empty result...

File: core/lspserver.py (hunk 1 above)
Comment (manateelazycat) [reply core/lspserver.py:350]: It's ok.

We can add https://microsoft.github.io/language-server-protocol/specification#textDocument_declaration and https://microsoft.github.io/language-server-protocol/specification#textDocument_typeDefinition later.

**Comment by manateelazycat** [comment]
> @manateelazycat I test with [this file](https://github.com/1history/1History/blob/74b5a1667c4f0e452beace35f8ce03fd6c64e5e8/src/database.rs#L26)
> 
> `lsp-bridge-find-define` works as expected, but not for completion. After typing `conn.`, `company-diag` return
> 
> ```
> Emacs 28.1 (aarch64-apple-darwin21.1.0) of 2022-04-05 on armbob.lan
> Company 0.9.13
> 
> company-backends: ((company-lsp-bridge))
> 
> Used backend: (company-lsp-bridge)
> 
> Major mode: rust-mode
> Prefix: ""
> Completions: none
> ```
> 
> In `*lsp-bridge*` buffer, rust-analyzer return 174 completion items, I have no idea why company got empty result...

已经问题 https://emacs-china.org/t/corfu-all-the-icons-icon/20907/7?u=manateelazycat

lsp-bridge 是完全异步的，其实补全的东西已经返回并记录到 lsp-bridge-completion-items 变量中，但是 company 和 corfu 弹出比较快，我还没有研究出怎么异步让 company 更新菜单内容的方法。

**Comment by jiacai2050** [comment]
> 我还没有研究出怎么异步让 company 更新菜单内容的方法。

感觉可以去看看 eglot 的实现，eglot 就工作的很好，补全也不卡。

**Comment by manateelazycat** [comment]
> 

这个问题肯定可以解决的，这几天写 lsp-bridge 架构太累了，我休息几天再看。

**Pull request merged**